.github/workflows/docker.yml

```diff
@@ -0,0 +1,105 @@
+name: Build and publish Docker images
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - "main"
+  workflow_run:
+    workflows: [Rebase Upstream]
+    types:
+      - completed
+
+jobs:
+  docker:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - id: cu118
+            name: CUDA 11.8
+            pytorch_install_args: "--index-url https://download.pytorch.org/whl/cu118"
+          - id: cu121
+            name: CUDA 12.1
+            pytorch_install_args: "--index-url https://download.pytorch.org/whl/cu121"
+          - id: cu124
+            name: CUDA 12.4
+            pytorch_install_args: "--index-url https://download.pytorch.org/whl/cu124"
+          - id: rocm6.2
+            name: ROCm 6.2
+            pytorch_install_args: "--index-url https://download.pytorch.org/whl/rocm6.2"
+          - id: cpu
+            name: CPU only
+            pytorch_install_args: "--index-url https://download.pytorch.org/whl/cpu"
+            extra_args: --cpu
+
+
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Check which repositories to use
+        id: repositories
+        run: |
+          echo "GHCR_IMAGE_NAME=ghcr.io/${GITHUB_REPOSITORY_OWNER}/comfyui" >> "$GITHUB_ENV"
+          if [[ -n "${DOCKERHUB_USERNAME}" ]]; then
+            echo "DOCKERHUB_IMAGE_NAME=${DOCKERHUB_USERNAME}/comfyui" >> "$GITHUB_ENV"
```

Comment:
I wouldn't push to both unless requested. I bloat the PR. Having an image on ghcr.io would be more than enough for me.

Reply:
DockerHub is pretty standard to publish to. GHCR is nice, publishing to both registries is low effort, I don't see it as bloat tbh.

I don't see the point of this step however to conditionally handle something when it shouldn't need to be dynamic. Either the PR is accepted and the credentials as secrets configured by maintainers or those registries aren't merged with the PR 🤷‍♂️

Reply:
You don't need to create a docker hub account and copy secrets to the repo config. So setting up ghcr.io requires significantly less effort of a maintainer. It's a simple as merging the PR.

Reply:
I don't see one or the other being the obvious choice, so I would prefer maintainer feedback on this, if one of the options should be removed and/or made mandatory.
Right now it's "one size fits all".

Reply:
Well, let's do it then, shall we? Have you pinged someone on elements or Discord? Apparently you are supposed to, according to the contributing guide.

Reply:
No, and I wasn't even aware, given that those contributing guidelines are newer than this PR 😂  Usually pinging someone about PRs is not very nice behavior, so I would never have guessed... I'll see if I can do this next week.